src/mlpack/core/data/load_impl.hpp

```diff
@@ -164,10 +165,18 @@ bool Load(const std::string& filename,
 
   // We can't use the stream if the type is HDF5.
   bool success;
-  if (loadType != arma::hdf5_binary)
-    success = matrix.load(stream, loadType);
+
+  if (loadType != file_type::mlp_hdf5_binary)
```

Comment:
I would suggest a different strategy here.  I would check to see if `MatType` is an Armadillo type, and if it is, then convert `loadType` to an `arma::file_type` with `ToArmaFileType()` and then use Armadillo's loader.

It is really the overload of `data::Load()` with a `DatasetMapper<MapPolicy>()` where we will always need to use the custom implementation in `LoadCSVV()`.

(I am not sure if you are already planning to do what I suggested, and are just testing here.  If so, sorry for the unnecessary comment! :))

Reply:
Yea, we were just testing here. Actually, we might have to refactor `Load` function a bit.
@shrit might want to add more here?